functions/src/extract.ts

```diff
@@ -65,7 +65,13 @@ export default function extract(snapshot: DocumentSnapshot): object {
   // Check payload size and make sure its within limits before sending for indexing
   const payload = getPayload(snapshot);
   if (getObjectSizeInBytes(payload) < PAYLOAD_MAX_SIZE) {
-    return payload;
+    return {
+      ...payload,
+      lastmodified: {
+        _operation: 'IncrementSet',
+        value: Date.now(),
```

Comment:
IIRC the firebase objects themselves have a timestamp before it gets unwrapped, could it be used?

Reply:
yes, thanks for the suggestion and it was confirmed by the Firebase team to use the same.  Code is updated.